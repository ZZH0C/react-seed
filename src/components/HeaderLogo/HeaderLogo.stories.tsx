import { HeaderLogo } from './HeaderLogo';import { ComponentMeta, ComponentStory } from '@storybook/react';import React from 'react';export default {  title: 'UI/HeaderLogo',  component: HeaderLogo,} as ComponentMeta<typeof HeaderLogo>;const Template: ComponentStory<typeof HeaderLogo> = () => <HeaderLogo />;export const Primary = Template.bind({});