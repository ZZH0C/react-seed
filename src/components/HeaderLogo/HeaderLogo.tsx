import React from 'react';

interface HeaderLogoProps extends React.HTMLAttributes<HTMLElement> {
  desktopImgSrc?: string;
  mobileImgSrc?: string;
  desktopImgAlt?: string;
  mobileImgAlc?: string;
}

export const HeaderLogo: React.FC<HeaderLogoProps> = ({
  desktopImgAlt,
  desktopImgSrc,
  mobileImgSrc,
  mobileImgAlc,
}) => {
  return (
    <div className="logoBlock">
      <a href="/">
        <img
          alt={desktopImgAlt}
          src={desktopImgSrc}
          className="logo"
          title="Logo"
        />
        <img alt={mobileImgAlc} src={mobileImgSrc} className="logoMobile" />
      </a>
    </div>
  );
};
