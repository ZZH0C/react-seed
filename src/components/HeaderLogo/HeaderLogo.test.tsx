import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });
import { HeaderLogo } from './HeaderLogo';

describe('components/Head', () => {
  const mockProps = {
    desktopImgAlt: 'Image Alt',
    desktopImgSrc: '/fake-image-url',
    mobileImgAlc: 'Image Alt',
    mobileImgSrc: '/fake-image-url',
    children: 'mock children',
  };

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render and match snapshot', () => {
    expect(
      shallow(<HeaderLogo {...mockProps}>{mockProps.children}</HeaderLogo>),
    ).toMatchSnapshot();
  });
});
