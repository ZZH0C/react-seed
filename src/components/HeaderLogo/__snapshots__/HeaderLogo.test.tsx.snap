// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Head should render and match snapshot 1`] = `
<div
  className="logoBlock"
>
  <a
    href="/"
  >
    <img
      alt="Image Alt"
      className="logo"
      src="/fake-image-url"
      title="Logo"
    />
    <img
      alt="Image Alt"
      className="logoMobile"
      src="/fake-image-url"
    />
  </a>
  <span>
    mock children
  </span>
</div>
`;
